### DISCUSSION TITLE
why some props generate to html ?

### DISCUSSION THREAD

**Discussion by liyefox** [ben-rogerson/twin.macro#535] [Q&A] [answered]
`<FlexBox direction="column" tw="h-screen">`
generate
`<div data-tw="h-screen" class="css-o9a5n8-FlexBox-MainLayout e1k2islv1" style="" direction="column">`

with `direction="column"` in html tag

**Comment by ben-rogerson** [comment, marked as answer]
That's how the css-in-js library works under the hood (not a twin thing).
Generally, it will hash your style set, add it to a className and move the styles into the head.

If you're after more info on how it works you can [read up here](https://medium.com/styled-components/how-styled-components-works-618a69970421).
